ee/session_recordings/playlist_counters/recordings_that_match_playlist_filters.py

```diff
@@ -278,10 +286,71 @@ def convert_filters_to_recordings_query(playlist: SessionRecordingPlaylist) -> R
         raise
 
 
+def try_to_store_error_count(playlist_short_id: str | None) -> None:
+    try:
+        if not playlist_short_id:
+            return
+
+        redis_client = get_client()
+
+        existing_value = redis_client.get(f"{PLAYLIST_COUNT_REDIS_PREFIX}{playlist_short_id}")
+        if existing_value:
+            existing_value = json.loads(existing_value)
+        else:
+            existing_value = {}
+
+        error_date = datetime.now()
+        value_to_set = json.dumps(
+            {
+                **existing_value,
+                "errored_at": error_date.isoformat(),
+                "error_count": existing_value.get("error_count", 0) + 1,
+            }
+        )
+
+        redis_client.setex(
+            f"{PLAYLIST_COUNT_REDIS_PREFIX}{playlist_short_id}", THIRTY_SIX_HOURS_IN_SECONDS, value_to_set
+        )
+    except Exception:
+        REPLAY_PLAYLIST_COULD_NOT_ADD_ERROR_COUNT.inc()
+        pass
+
+
+def should_skip_task(existing_value: dict[str, Any], playlist_filters: dict[str, Any]) -> bool:
+    # if we have results from the last hour we don't need to run the query
+    if existing_value.get("refreshed_at"):
+        last_refreshed_at = datetime.fromisoformat(existing_value["refreshed_at"])
+        seconds_since_refresh = int((datetime.now() - last_refreshed_at).total_seconds())
+
+        if seconds_since_refresh <= settings.PLAYLIST_COUNTER_PROCESSING_COOLDOWN_SECONDS:
+            REPLAY_TEAM_PLAYLIST_COUNT_SKIPPED.labels(reason="cooldown").inc()
+            return True
+
+    # don't retry for a while if we're getting errors
+    if existing_value.get("errored_at"):
+        last_errored_at = datetime.fromisoformat(existing_value["errored_at"])
+        seconds_since_refresh = int((datetime.now() - last_errored_at).total_seconds())
+
+        if seconds_since_refresh <= settings.PLAYLIST_COUNTER_PROCESSING_COOLDOWN_SECONDS:
+            REPLAY_TEAM_PLAYLIST_COUNT_SKIPPED.labels(reason="error_cooldown").inc()
+            return True
+
+    # don't keep retrying if we keep getting errors
+    if existing_value.get("error_count", 0) >= 5:
+        REPLAY_TEAM_PLAYLIST_COUNT_SKIPPED.labels(reason="max_error_cooldown").inc()
+        return True
+
+    # if this is the default filters, then we shouldn't have allowed this to be created - we can skip it
+    if playlist_filters == DEFAULT_RECORDING_FILTERS:
+        REPLAY_TEAM_PLAYLIST_COUNT_SKIPPED.labels(reason="default_filters").inc()
+        return True
+
+    return False
+
+
 @shared_task(
     ignore_result=True,
     queue=CeleryQueue.SESSION_REPLAY_GENERAL.value,
-    # limit how many run per worker instance - if we have 10 workers, this will run 600 times per hour
     rate_limit="120/h",
```

Comment:
It may make sense to limit not 120/h but to like: 2/m, this would ensure the load is smoother over hour

Reply:
```suggestion
    rate_limit="2/m",
```